broker-core/src/main/java/io/zeebe/broker/subscription/message/state/WorkflowInstanceSubscriptionState.java

```diff
@@ -133,6 +130,7 @@ public void updateToClosingState(final WorkflowInstanceSubscription subscription
   }
 
   public void updateSentTime(final WorkflowInstanceSubscription subscription, long sentTime) {
+    // this is called from an pending checker which needs a transaction
```

Comment:
Wouldn't it be better to put this transaction in the pending checker? Here you don't know whether you're in a transaction or not. This is also called from other places like the above update methods which are called by processors so there's the issue you mentioned with the nested transactions. You'd have to expose the dbContext though, so I'm not sure if it's that much better

Reply:
You're right.